scripts/clang-build-analyzer

```diff
@@ -0,0 +1,40 @@
+#! /bin/bash
+
+# This script runs ClangBuildAanlyzer on a clean build of VAST
+#   Usage: ./scripts/clang-build-analyzer [<additional-configure-flags>]
+
+set -e
+
+# Create temporary directory
+WORKING_DIR=$(mktemp -d)
+
+# Build ClangBuildAnalyzer
+git clone "https://github.com/aras-p/ClangBuildAnalyzer.git" \
+  "${WORKING_DIR}/CBA"
+pushd "${WORKING_DIR}/CBA"
+make -f "projects/make/Makefile"
+popd
+CBA="${WORKING_DIR}/CBA/build/ClangBuildAnalyzer"
+
+# Configure VAST to build with time-traces enabled
+./configure "${@}" --build-dir="${WORKING_DIR}/VAST" \
+  --extra-flags="-ftime-trace"
+
+# When not on master, build first, then touch the changed files
+if [ "$(git rev-parse --abbrev-ref HEAD)" != "master" ]; then
+  cmake --build "${WORKING_DIR}/VAST" -j \
+    | grep -v "Time trace\|tracing"
+  git diff --name-only "$(git merge-base origin/master HEAD)" \
```

Comment:
It would be nice to replace `"$(git merge-base origin/master HEAD)"` with `"$BASE"` and define `BASE` as `BASE="${1:=HEAD}"` instead.

Reply:
I've decided to use the variable `CBA_BASE` instead, defaulting to `HEAD`.